oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/AbstractOAuth2Request.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.auth.oauth2;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.QueryParams;
+import com.linecorp.armeria.common.QueryParamsBuilder;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.RequestHeadersBuilder;
+
+/**
+ * A common abstraction for the requests implementing various OAuth 2.0 request/response flows,
+ * as per <a href="https://tools.ietf.org/html/rfc6749">[RFC6749]</a> and other relevant specifications.
+ * @param <T> the type of the authorization result.
+ */
+public abstract class AbstractOAuth2Request<T> {
```

Comment:
As I mentioned, the `*Request` classes in `common.auth.oauth2` could be hidden from the public API by moving them to `internal.common.auth.oauth2`.

Reply:
moved to internal as suggested

Reply:
Thanks, @max904-github. Will review again soon.